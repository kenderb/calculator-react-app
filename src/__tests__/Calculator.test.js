import { create } from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../components/Calculator';

test('renders correctly', () => {
  const tree = create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('displays heading', () => {
  render(<Calculator />);
  const h1 = screen.getByText('Let’s do some math!');
  expect(h1).toBeInTheDocument();
});

test('displays numbers', () => {
  render(<Calculator />);
  const button = screen.getByText('2');
  expect(button).toBeInTheDocument();
});

test('displays operator', () => {
  render(<Calculator />);
  const button = screen.getByText('+');
  expect(button).toBeInTheDocument();
});

test('displays correct result', () => {
  render(<Calculator />);
  fireEvent.click(screen.getByText('1'));
  fireEvent.click(screen.getByText('+'));
  fireEvent.click(screen.getByText('9'));
  fireEvent.click(screen.getByText('='));
  expect(screen.getByText('10')).toBeInTheDocument();
});

test('displays correct result on division', () => {
  render(<Calculator />);
  fireEvent.click(screen.getByText('1'));
  fireEvent.click(screen.getByText('÷'));
  fireEvent.click(screen.getByText('2'));
  fireEvent.click(screen.getByText('='));
  expect(screen.getByText('1')).toBeInTheDocument();
});
