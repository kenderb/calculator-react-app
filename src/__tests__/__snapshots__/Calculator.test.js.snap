// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gKsewC hYMqTz"
>
  <h1>
    Let’s do some math!
  </h1>
  <div
    className="sc-iBPRYJ gicZfC"
  >
    <div
      className="sc-bdfBwQ hXDNAS"
    >
      0
    </div>
    <div
      className="sc-gsTCUz dYdny"
    >
      <div>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="sc-dlfnbm kLtawK"
          color="true"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="sc-dlfnbm kLtawK"
          color="true"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="sc-dlfnbm kLtawK"
          color="true"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="sc-dlfnbm kLtawK"
          color="true"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div>
        <button
          className="sc-dlfnbm fbeBMw"
          color="false"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="sc-dlfnbm fLGXYK"
          color="false"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="sc-dlfnbm kLtawK"
          color="true"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
